test-integration/jdl-samples/ms-ng-oauth2-mongodb-caffeine/blog-store.jdl

```diff
@@ -16,41 +16,44 @@
  * - buildTool: maven, gradle
  */
 
-/* reactive, maven */
+/* imperative, maven */
 application {
   config {
     applicationType gateway
     authenticationType oauth2
     baseName gateway
     buildTool maven
+    cacheProvider caffeine
     clientFramework angular
     creationTimestamp 1617901618886
     databaseType mongodb
     jhiPrefix custom
     jwtSecretKey "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ="
     messageBroker kafka
     packageName com.okta.developer.gateway
-    serviceDiscoveryType consul
+    reactive false
+    serviceDiscoveryType no
```

Comment:
Gateway shouldn't be an option for microservice architecture, there's no point specifically routing.

I'm strongly against adding an antipattern just to please users that aren't able to achieve reactive programming from api to ms webclient.

Reply:
I agree that hard-coding routes in the gateway might not be a good idea. The reason we're considering it is that Spring Cloud Gateway MVC won't have support for service discovery until November 2024. 

As far as reactive vs MVC, there does seem to be a demand for Spring Cloud Gateway MVC, that's why Spencer Gibb and the Spring Cloud team are working on it.

Reply:
I don't agree.
- it's not up to us to say `you should use service discovery for every gateway/microservice use case`
- we do not forbid gateways and microservices without serviceDiscovery
- https://github.com/jhipster/generator-jhipster/issues/21012 is open for more than a year without any comment

IMO we have do choose one of:
- accept manual routes PR
- force the user to use `--skip-checks` for gateways and microservices without serviceDiscovery and close https://github.com/jhipster/generator-jhipster/issues/21012

Reply:
It's just IMHO:

Backend Gateways:
- Act as a single point of entry of frontend
- Is in charge of managing permissions (users and authorities)
- Routing ALL frontend Requests to the according microservice by querying the registry and translating exceptions and return codes
- Retry, circuitbreaking policy, ...
And this is simple enough to be managed by reactive paradigm to anyone (imperative would also please me as I find reactive quite hard, but still)

Otherwise, we loose one of the 12 factors and microservice paradigm. Manual route is not hipsterish and definitely not something worth the maintenance, I would personally prefer to close the #21012 as antipattern

Reply:
ideally BTW, this paradigm should also be applied for microfrontend too ;-)

Reply:
To add to this discussion, here's a video from Josh Long about why Spring Cloud Gateway MVC. Summary: because Java 21 with virtual threads and it's easier to understand. https://youtu.be/1ouE2QAebuE?si=-603Akrcvgn0HiMN&t=244